app/controllers/seances_controller.rb

```diff
@@ -1,13 +1,11 @@
 class SeancesController < ApplicationController
-  before_action :set_seance, only: %i[show edit update destroy]
+  include Pundit::Authorization
 
-  # GET /seances or /seances.json
-  def index
-    @seances = Seance.all
-  end
+  rescue_from Pundit::NotAuthorizedError, with: :user_not_authorized
+  # Probably should move this after implementing errorhandler
 
-  # GET /seances/1 or /seances/1.json
-  def show; end
+  before_action :set_seance, only: %i[edit update destroy]
+  before_action :manager_authenticate
```

Comment:
Ok, so that's not how Pundit is supposed to be used. If you would create policies per resource, as in comment above, you then can have policy like that(let's just use two actions - create and update - as an example, rest of them will be similar):

```ruby
class SeancePolicy < ApplicationPolicy
  def update?
    manager?
  end

  def create?
    manager?
  end

  private

  def manager?
    user.role == "manager"
  end
end
```

With that defined policy we can have controller like that:

```ruby
class SeancesController < ApplicationController
  include Pundit::Authorization

  before_action :set_seance, only: %i[update]
  before_action :authenticate_user!

  # POST /seances or /seances.json
  def create
    authorize Seance
    @seance = Seance.new(seance_params)
    respond_to do |format|
      if @seance.save
        format.html { redirect_to seance_url(@seance), notice: "Seance was successfully created." }
        format.json { render :show, status: :created, location: @seance }
      else
        format.html { render :new, status: :unprocessable_entity }
        format.json { render json: @seance.errors, status: :unprocessable_entity }
      end
    end
  end

  # PATCH/PUT /seances/1 or /seances/1.json
  def update
    respond_to do |format|
      if @seance.update(seance_params)
        format.html { redirect_to seance_url(@seance), notice: "Seance was successfully updated." }
        format.json { render :show, status: :ok, location: @seance }
      else
        format.html { render :edit, status: :unprocessable_entity }
        format.json { render json: @seance.errors, status: :unprocessable_entity }
      end
    end
  end

  private
  # Use callbacks to share common setup or constraints between actions.
  def set_seance
    @seance = authorize Seance.find(params[:id])
  end
end
```

This will now check in `SeancePolicy` whenever logged in user can perform create/update, which in turns check if that user is manager. Pundit will know which method to call based on action name in controller, and which Policy to use based on class of the record 😉